src/main/java/net/rptools/maptool/events/ZoneLoadedListener.java

```diff
@@ -14,28 +14,49 @@
  */
 package net.rptools.maptool.events;
 
-import static net.rptools.maptool.client.functions.MapFunctions.ON_CHANGE_MAP_CALLBACK;
-
 import com.google.common.eventbus.Subscribe;
 import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import net.rptools.maptool.client.MapTool;
 import net.rptools.maptool.client.events.ZoneLoaded;
-import net.rptools.maptool.model.Token;
+import net.rptools.maptool.client.ui.zone.ZoneRenderer;
+import net.rptools.maptool.language.I18N;
+import net.rptools.maptool.model.library.Library;
+import net.rptools.maptool.model.library.LibraryManager;
 import net.rptools.maptool.util.EventMacroUtil;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 
 public class ZoneLoadedListener {
+  private static final Logger LOGGER = LogManager.getLogger(EventMacroUtil.class);
+  public static final String ON_CHANGE_MAP_CALLBACK = "onChangeMap";
 
   public ZoneLoadedListener() {
     new MapToolEventBus().getMainEventBus().register(this);
   }
 
   @Subscribe
   public void OnChangedMap(ZoneLoaded event) {
-    var libTokens = EventMacroUtil.getEventMacroTokens(ON_CHANGE_MAP_CALLBACK);
-    String prefix = ON_CHANGE_MAP_CALLBACK + "@";
-
-    for (Token handler : libTokens) {
-      EventMacroUtil.callEventHandlerOld(
-          prefix + handler.getName(), "", handler, Collections.emptyMap(), true);
+    ZoneRenderer currentZR = MapTool.getFrame().getCurrentZoneRenderer();
+    try {
+      var libs = new LibraryManager().getLegacyEventTargets(ON_CHANGE_MAP_CALLBACK).get();
+      if (libs.isEmpty()) return;
+      for (Library handler : libs) {
+        try {
+          String libraryNamespace = handler.getNamespace().get();
+          EventMacroUtil.callEventHandler(
+              ON_CHANGE_MAP_CALLBACK,
+              libraryNamespace,
+              currentZR.getZone().getId().toString() + "," + currentZR.getZone().toString(),
```

Comment:
I think we should remove the second argument as it's not needed - if the event handler does need it, it can get it based on the ID.

The result of `Zone#toString()` is also not very predictable: depending on whether it's a GM or player client, and whether the name and display name match, there could be several possible values. So if we keep the parameter, we should pick either `getName()` or `getDisplayName()` for it.

Reply:
I glanced at the code and don't see an answer — why is the zone name being used instead of the id in the first place?  If it's just informational, why not just the zone id and let the listener query other attributes that it wants?

Reply:
I assumingly thought there would be added value to include the _textual name_ as it was readily available.
I will remove it.